test/e2e/operator_test.go

```diff
@@ -24,3 +33,136 @@ func TestOperatorNamespace(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestPodDisruptionBudgetAtLimitAlert tests that pdb-atlimit alert exists when there is a pdb at limit
+// See https://bugzilla.redhat.com/show_bug.cgi?id=1762888
+func TestPodDisruptionBudgetAtLimitAlert(t *testing.T) {
+	kubeConfig, err := test.NewClientConfigForTest()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	policyClient, err := policyclientv1beta1.NewForConfig(kubeConfig)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	labels := map[string]string{"app": "pdbtest"}
+	name := names.SimpleNameGenerator.GenerateName("pdbtest-")
+	cmName := names.SimpleNameGenerator.GenerateName("test-configmap-")
+
+	ns, err := createTestNamespace(kubeClient, name)
+	if err != nil {
+		t.Fatalf("could not create test namespace: %v", err)
+	}
+	defer kubeClient.CoreV1().Namespaces().Delete(ns.Name, &metav1.DeleteOptions{})
+
+	err = podCreate(kubeClient, name, labels)
+	if err != nil {
+		t.Fatal(err)
+	}
+	var pods *corev1.PodList
+	testTimeout := time.Second * 120
+	// Poll to see if pod is created
+	wait.PollImmediate(time.Second*1, testTimeout, func() (bool, error) {
+		pods, err = kubeClient.CoreV1().Pods(name).List(metav1.ListOptions{LabelSelector: "app=pdbtest"})
+		if err != nil {
+			return false, err
+		}
+		if len(pods.Items) > 0 && pods.Items[0].Status.Phase == corev1.PodRunning {
+			return true, nil
+		}
+		return false, nil
+	})
+
+	err = pdbCreate(policyClient, name, labels)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Wait for pbd
+	err = wait.PollImmediate(time.Second*1, testTimeout, func() (bool, error) {
+		if _, err := policyClient.PodDisruptionBudgets(name).List(metav1.ListOptions{LabelSelector: "app=pbtest"}); err != nil {
+			return false, fmt.Errorf("Waiting for poddisruptionbudget")
+		}
+		return true, nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	caBundle, err := metrics.RetrieveServiceCABundle(kubeClient, cmName, ns.Name)
```

Comment:
I'm happy to see this decomposed implementation - I think it will prove easier to maintain. I think the next step is adding a helper method to simplify calling for cases where only a single query is required e.g.

```golang
metrics.QueryPrometheus(kubeClient, ns.Name, query)
```